db/migrate/20210930172204_add_support_for_compound_samples.rb

```diff
@@ -0,0 +1,13 @@
+# frozen_string_literal: true
+
+# Add a join table to allow samples to be parents and children of each other.
+class AddSupportForCompoundSamples < ActiveRecord::Migration[6.0]
+  def change
+    create_table 'compound_samples' do |t|
+      t.integer 'parent_id', null: false
+      t.integer 'child_id', null: false
```

Comment:
See comment in the test file, regarding naming. Thanks!

Reply:
I'm going to rename the join type from `ComponentSample` to `SampleRelationship` as I don't think it's clear what's meant by having a column in the `compound_samples` table called `compound_sample_id` when there's also an `id` that's the ID of the object with the same name.

Reply:
I was imagining the join table would be called the generic name, like SampleLink or SampleRelationship. 
Then the rails relationships could be called compound_samples, and component_samples, and the id fields on the table called `compound_sample_id` and `component_sample_id`.
Is that the same as what you're suggesting?

Reply:
Yes I just did a terrible job of explaining it, but what you described seems to match with what I was trying to 😁

Reply:
haha OK great, glad we agree!